### PR TITLE
SPDY roundtripper support to proxy with Basic auth

### PR DESCRIPTION
**Pull request by fabianofranz**
Adding support to http proxy servers with Basic auth to the SPDY roundtripper. Fixes `kubectl exec` when under a proxy that requires authentication.

### CODE REVIEW

**Comment by fabianofranz** [comment]
@liggitt PTAL. Note this PR depends/includes the commit in https://github.com/kubernetes/kubernetes/pull/21185.

**Comment by liggitt** [comment]
LGTM

File: pkg/util/httpstream/spdy/roundtripper_test.go [hunk 1]
```diff
@@ -146,23 +264,56 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 			stream := <-streamCh
 			io.Copy(stream, stream)
 		}))
-		// TODO: Uncomment when fix #19254
-		// defer server.Close()
+		defer server.Close()
```
Comment (liggitt) [inline pkg/util/httpstream/spdy/roundtripper_test.go]: not sure about this.. seeing timeouts on some unit tests downstream on this

Comment (liggitt) [inline pkg/util/httpstream/spdy/roundtripper_test.go]: yeah, https://github.com/kubernetes/kubernetes/issues/19254#issuecomment-174734626 was closed not with a fix, but by commenting out the .Close() calls

Comment (fabianofranz) [inline pkg/util/httpstream/spdy/roundtripper_test.go]: Commented again. Let's see how the downstream tests go.

**Comment by liggitt** [comment]
tests look good

**Comment by fabianofranz** [comment]
@liggitt had to rebase.

**Pull request merged**